internal/k8s/builder/gateway.go

```diff
@@ -81,11 +80,12 @@ func filterAnnotations(annotations map[string]string, allowed []string) map[stri
 }
 
 type GatewayDeploymentBuilder struct {
-	gateway      *gw.Gateway
-	gwConfig     *v1alpha1.GatewayClassConfig
-	sdsHost      string
-	sdsPort      int
-	consulCAData string
+	gateway          *gw.Gateway
+	gatewayNamespace string
+	gwConfig         *v1alpha1.GatewayClassConfig
+	sdsHost          string
+	sdsPort          int
+	consulCAData     string
```

Comment:
```suggestion
	gateway                *gw.Gateway
	gwConfig               *v1alpha1.GatewayClassConfig
	sdsHost                string
	sdsPort                int
	consulCAData           string
	consulGatewayNamespace string
```
For places if/where we're referencing Consul namespaces instead of k8s namespaces, I think it would be really helpful to clarify that in variable/struct field naming. (I'm not sure if this is actually one of those places though.)